### PR TITLE
Links in `material` library docs are outdated

### PR DESCRIPTION
**Pull request by ttanimichi**
https://api.flutter.dev/flutter/material/material-library.html

Links in `material` library docs are outdated

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Review by justinmc — APPROVED** [review]
LGTM 👍 

Thanks for taking the time to correct this!

**Review by loic-sharma — APPROVED** [review]

File: packages/flutter/lib/material.dart [hunk 1]
```diff
@@ -10,14 +10,10 @@
 ///
 /// See also:
 ///
-///  * [flutter.dev/widgets/material](https://flutter.dev/widgets/material)
+///  * [docs.flutter.dev/ui/widgets/material](https://docs.flutter.dev/ui/widgets/material)
 ///    for a catalog of commonly-used Material component widgets.
-///  * [material.io/design](https://material.io/design/)
+///  * [m3.material.io](https://m3.material.io/)
```
Comment (guidezpl) [inline packages/flutter/lib/material.dart:15]:
```suggestion
///  * [m3.material.io](https://m3.material.io/) for the Material 3 specification
///  * [m2.material.io](https://m2.material.io/) for the Material 2 specification
```

**Review by guidezpl — CHANGES_REQUESTED** [review]
I think we still want to link to m2 docs as long as the library supports it (related: https://github.com/flutter/flutter/issues/127064)

File: packages/flutter/lib/material.dart (hunk 1 above)
Comment (ttanimichi) [reply packages/flutter/lib/material.dart]: @guidezpl Fixed https://github.com/flutter/flutter/pull/129891/commits/b3b435221dbce09e9d302db9529f5c2d47d9b961 and https://github.com/flutter/flutter/pull/129891/commits/535d3c8932ab301c92c52338ea69805559944bc4

**Review by guidezpl — APPROVED** [review]

**Pull request merged**